frontend/src/app/main/ui/ds/buttons/icon_button.cljs

```diff
@@ -0,0 +1,30 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.buttons.icon-button
+  (:require-macros
+   [app.common.data.macros :as dm]
+   [app.main.style :as stl])
+  (:require
+   [app.main.ui.ds.foundations.assets.icon :refer [icon*]]
+   [rumext.v2 :as mf]))
+
+(def button-variants (set '("primary" "secondary" "ghost" "destructive")))
+
+(mf/defc icon-button*
+  {::mf/props :obj}
+  [{:keys [class icon variant aria-label] :rest props}]
+  (when (some? variant) (assert (contains? button-variants variant) "Invalid variant"))
+  (assert (some? aria-label) "aria-label must be provided")
+  (assert (some? icon) "An icon ID must be provided")
```

Comment:
Just for consistency with the rest of assertions, the message can be all lowercase?